### PR TITLE
Update maven-invoker-plugin to 3.4.0 and add to dependabot

### PR DESCRIPTION
**Pull request by famod**
https://github.com/apache/maven-invoker-plugin/releases

Although it now brings a recent groovy version (4.0.9), I kept the explicit groovy deps as per https://github.com/quarkusio/quarkus/pull/31301#issuecomment-1437595319.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -79,6 +79,7 @@ updates:
       - dependency-name: io.fabric8:docker-maven-plugin
       - dependency-name: net.revelc.code.formatter:formatter-maven-plugin
       - dependency-name: net.revelc.code:impsort-maven-plugin
+      - dependency-name: org.apache.maven.plugins:maven-invoker-plugin
```
Comment (famod) [inline .github/dependabot.yml:82]: I do think we should start adding more standard Maven plugins.
Even for compiler-plugin, which often requires additional adjustments for codestarts, it's still worthwhile to know when we are falling behind.

Comment (gastaldi) [reply .github/dependabot.yml:82]: Sure, why not?

Comment (famod) [reply .github/dependabot.yml:82]: That would be a separate PR then 🙂

Comment (gastaldi) [reply .github/dependabot.yml:82]: Of course 😉

**Comment by famod** [comment]
It seems additional adjustments are needed. Converted to draft for now.

**Comment by famod** [comment]
I had to step back to 3.4.0 due to https://issues.apache.org/jira/browse/MINVOKER-328 (which will be part of 3.5.1).

After merging, dependabot will probably create a pr updating to 3.5.0. We'll have to close that then.

**Review by gsmet — APPROVED** [review]

**Review by gastaldi — APPROVED** [review]

**Comment by famod** [comment]
CI finished with a few timeouts, but none look related. CI was fully green in my fork. I'll go ahead and merge this.

**Pull request merged**